File: apina-model/src/main/java/fi/evident/apina/model/parameters/EndpointRequestParamParameter.java
package fi.evident.apina.model.parameters;

import fi.evident.apina.model.type.ApiType;

import java.util.Optional;

public final class EndpointRequestParamParameter extends EndpointParameter {

    private final String queryParameter;

    public EndpointRequestParamParameter(String name, Optional<String> queryParameter, ApiType type) {
        super(name, type);

        this.queryParameter = queryParameter.orElse(name);
    }

    public String getQueryParameter() {
        return queryParameter;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("@RequestParam");

        if (!queryParameter.equals(name))
            sb.append("(\"").append(queryParameter).append("\")");

        sb.append(' ').append(type).append(' ').append(name);
        return sb.toString();
    }
}

File: apina-core/src/main/kotlin/fi/evident/apina/model/parameters/EndpointRequestParamParameter.kt
package fi.evident.apina.model.parameters

import fi.evident.apina.model.type.ApiType

class EndpointRequestParamParameter(name: String, queryParameter: String?, type: ApiType) : EndpointParameter(name, type) {

    val queryParameter: String = queryParameter ?: name

    override fun toString(): String = buildString {
        append("@RequestParam")

        if (queryParameter != name)
            append("(\"").append(queryParameter).append("\")")

        append(' ').append(type).append(' ').append(name)
    }
}
